modules/vstudio/tests/cs2005/test_dotnetsdk.lua

```diff
@@ -0,0 +1,121 @@
+--
+-- tests/actions/vstudio/cs2005/test_dotnetsdk.lua
+-- Test DotnetSDK feature Visual Studio 2005+ C# project.
+-- Copyright (c) 2012-2024 Jason Perkins and the Premake project
+--
+	local p = premake
+	local suite = test.declare("vstudio_cs2005_dotnetsdk")
+	local dn2005 = p.vstudio.dotnetbase
+--
+-- Setup
+--
+
+	local wks, prj
+
+--
+-- Setup and teardown
+--
+	function suite.setup()
+		p.action.set("vs2010")
+		wks = test.createWorkspace()
+	 	configurations { "Debug", "Release" }
+		language "C#"
+		dotnetframework "net8.0"
+	end
+
+	local function setConfig()
+		local cfg = test.getconfig(prj, "Debug")
+		dn2005.projectElement(cfg);
+	end
+
+	local function prepare()
+		prj = test.getproject(wks, 1)
+	end
+
+	function suite.testNone()
+		prepare()
+		setConfig()
+
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk">
+	]]
+	end
+
+	function suite.testDefault()
+		prepare()
+		setConfig()
+		dotnetsdk "Web"
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk">
+	]]
+	end
+
+	function suite.testWeb()
+		prepare()
+		dotnetsdk "Web"
+		setConfig()
+
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk.Web">
+		]]
+	end
+
+	function suite.testRazor()
+		prepare()
+		dotnetsdk "Razor"
+		setConfig()
+
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk.Razor">
+		]]
+	end
+
+	function suite.testWorker()
+		prepare()
+		dotnetsdk "Worker"
+		setConfig()
+
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk.Worker">
+		]]
+	end
+
+	function suite.testBlazor()
+		prepare()
+		dotnetsdk "Blazor"
+		setConfig()
+
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+		]]
+	end
+
+	function suite.testWindowsDesktop()
+		prepare()
+		dotnetsdk "WindowsDesktop"
+		setConfig()
+
+		test.capture [[
+<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+		]]
+	end
+	function suite.testMSTest()
```

Comment:
global.json part is not tested

Reply:
how would i go about that?

Reply:
Issue is probably that global.json doesn't follow other file generation and so `dotnetbase.netcore.dotnetsdk` would be misplaced...

Reply:
would you happen to know where i can find another example of this file generating?

Reply:
I though about `premake.generate`, something like:
```lua
local content = p.capture(function() generate_global_json_content(prj) end)
if content ~= nil and #content > 0 then
	p.generate(prj, "global.json", function() p.outln(content) end)
end
```
That way, `generate_global_json_content` can be tested.
User is notified of the creation of the file.

Reply:
i got premake.generate to work except the test is not working probably somthing stupid that i am missing!

Reply:
![image](https://github.com/user-attachments/assets/1a7573e2-7bc2-4ae3-b42e-80d675782f90)
i got this far but it seems the content is not getting captured properly